pkg/config/templates/multipath.conf.tmpl

```diff
@@ -0,0 +1,8 @@
+blacklist {
```

Comment:
Do we need a default `devnode` line here that blacklists everything? I was just looking at the manpage for multipath.conf, and it says:

> The default blacklist consists of the regular expression "!^(sd[a-z]|dasd[a-z]|nvme[0-9])". This causes all device types other than scsi, dasd, and nvme to be excluded from multipath handling by default."

My reading is that with multipathd enabled all scsi, dasd and nvme devices are still up for grabs. Is this correct, or am I confused?

Reply:
from my experiments we are blacklisting everything apart from the negated expression in blacklist which whitelists the devices.

for example on my kvm setup the following config.yaml is used

```
os:
  externalStorageConfig:
    enabled: true
    multiPathConfig:
    - vendor: "QEMU"
      product: "QEMU HARDDISK"
```

This gets converted into multipath config

```
blacklist {

    device {
        vendor "!QEMU"
        product "!QEMU HARDDISK"
    }

}
```

end result is multipath is enabled on the qemu scsi disks

```
0QEMU_QEMU_HARDDISK_disk1 dm-0 QEMU,QEMU HARDDISK
size=250G features='0' hwhandler='0' wp=rw
|-+- policy='service-time 0' prio=1 status=active
| `- 0:0:0:1 sda 8:0  active ready running
`-+- policy='service-time 0' prio=1 status=enabled
  `- 0:0:0:2 sdb 8:16 active ready running
```

Longhorn disks in use by VM's are ignored by mutlipath since they do not match the expression
```
  *-disk
       description: SCSI Disk
       product: VIRTUAL-DISK
       vendor: IET
       physical id: 0.0.1
       bus info: scsi@7:0.0.1
       logical name: /dev/sdc
       version: 0001
       serial: beaf11
       size: 10GiB (10GB)
       capabilities: gpt-1.00 partitioned partitioned:gpt
       configuration: ansiversion=5 guid=57debec6-c414-4d56-a0a1-788da7c89c52 logicalsectorsize=512 sectorsize=512
```